frontend/main/app/AppController.js

```diff
@@ -11,64 +7,76 @@ import { NavigationMenu } from '../components/NavigationMenu/NavigationMenu.js';
 import { ProfilePage } from '../components/ProfilePage/ProfilePage.js';
 import { SellProductsPage } from '../components/MarketplacePage/SellProductsPage.js';
 
-
 export class AppController {
    #container = null;
-   #currentView = null; // Track the currently rendered views
+   #currentView = null; // Track the currently rendered view
    #views = {}; // Store initialized views
-   #navigationMenu = null; // Store the navigation menu separately and add on views
+   #navigationMenu = null; // Navigation menu component
+   static instance = null; // Singleton instance
 
    constructor() {
-      // Initialize components
-      this.#views = {
-         marketplace: new MarketplacePage(),
-         secureCheckout: new SecureCheckout(),
-         virtualCart:  new VirtualCart(), 
-         navigationMenu: new NavigationMenu(),
-         productPage: new ProductPage(),
-         profilePage: new ProfilePage(),
-         sellProductsPage: new SellProductsPage()
-      };
+     // Initialize components
+     this.#views = {
+       marketplace: new MarketplacePage(),
+       secureCheckout: new SecureCheckout(),
+       virtualCart: new VirtualCart(),
+       navigationMenu: new NavigationMenu(),
+       productPage: new ProductPage(), // ProductPage is dynamically initialized when needed
+       profilePage: new ProfilePage(),
+       sellProductsPage: new SellProductsPage(),
+     };
 
-      // Default Page set as marketplace page
-      this.#currentView = this.#views.marketplace;
+     // Set the default page as the marketplace
+     this.#currentView = this.#views.marketplace;
    }
 
-  render() {
-    if (!this.#container) {
-      this.#container = document.createElement('div');
-      this.#container.classList.add('app-controller');
-    }
+   async render(id=null) {
```

Comment:
Why is render made asynchronous?

Reply:
Fetching data from DB is an async operation